### PR TITLE
[b/368572924] Fix: missing data of public synonym objects

### PR DESCRIPTION
**Pull request by misolt**
Fix the issue by removing the filter

### CODE REVIEW

**Review by shevek-google — COMMENTED** [review]
I randomly picked this PR to look at for no good reason, and my thought is: Would these be easier to review and/or would the code review process be more reliable if a PR like this came with a proof of behaviour on some Oracle version(s)?

The reviewer has less information than the developer, because they don't have the ability to test it, and to approve its behaviour, their only option is to detangle the code and try to figure it out.

Compare this to the TT system in CW, where every committed code change comes with a YAML file capturing its behaviour in detail, and that behaviour can be reviewed and approved, reducing the code review to a minimum.

Nothing I say here has any bearing on this PR in particular; this is not a review, as-such, but perhaps this is an inspiration to think about the question of using evidence to make reviews faster, easier and more useful, and put some ideas in the back of your mind.

Happy to talk about TT any time you like if that's interesting.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql
Comment (kpiotrowski) [inline dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql:22]: I think we shouldn't introduce TableOwner column to db-objects because it does not make sense in this context and won't be used by assessment. TableOwner is requires to filter synonyms only that are ingested to the separate temporary table. After the filtering assessment should take care about merging synonyms to db-objects.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql
Comment (kpiotrowski) [inline dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql:22]: I suggest to remove Count column from the synonyms sql. We can easily add this filtering on the assessment side. This will reduce dumper complexity.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql
Comment (misolt) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql:22]: The column supplies no data, but on the Assessment end we rely on these two CSVs having the same structure.

Even after the filtering, this column will be present in the tmp table, so we'll need to specify which columns are getting merged and which aren't. I think allowing the redundant column with nulls is a safer option.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql
Comment (misolt) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql:22]: This is a good idea, but it must wait until Assessment no longer relies on this column. Even though the information in the column can be obtained elsewhere, current-version Assessment could still try to read it and get errors.

Comment (kpiotrowski) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql:22]: How about we handle this condition in the same way as it is in db-objects.sql SQL and add `WHERE A.object_name NOT LIKE 'BIN$%' AND A.object_name NOT LIKE '/%'`? We can then keep simple `count(1)` for backward compatibility and remove it later after changing assessment.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql
Comment (misolt) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql:22]: I think either way this will be a non-trivial decision with some future consequences, so we should discuss it with the team before merging the PR or the CL.

Comment (dominik-klimczak) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql:22]: Let's not deviate from the dumper design principles that the tool should be as simple and as reliable as possible. It's better to add extra processing in the service than increase the complexity here.

Comment (vladislav-sidorovich) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql]: Colleagues, I see that we agreed on the statements:
1. Keep dumper as simple as possible
2. Add additional logic (complexity) on server if needed

Could you please clarify what specific actions follow from this?

Comment (dominik-klimczak) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql]: The discussion also took place in the ingestion code. The outcome is we don't add a null-filled column "TableOwner" in the dumper.

Comment (misolt) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects.sql]: The discussion here is a little out of date, the PR is currently on hold because we can't make a release at the moment.

I'll update it before reopening, but it's as Dominik said, the TableOwner will only be present in the synonym-public version of DB Objects.

File: dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql
Comment (misolt) [reply dumper/app/src/main/resources/oracle-stats/cdb/native/db-objects-synonym-public.sql]: Done, I used the constant `1` instead of counting the items, because without grouping by name the number won't be useful.

**Review by kpiotrowski — APPROVED** [review]

**Pull request merged**
